### DISCUSSION TITLE
API endpoints for Video Generation

### DISCUSSION THREAD

**Discussion by ncrovatti** [thygate/stable-diffusion-webui-depthmap-script#310] [Ideas]
Hello folks, 

This project is awesome, can't get enough of it. One feature that is available in the extension UI is the ability to generate Generate 3D inpainted mesh, once the depth map object is generated, it's possible to generate videos from it and it's a superb functionality !  

What would be awesome, is to be able to access that particular set of features from API endpoints to build websites and applications. 

Wondering about your thought. 

Cheers,
Nicolas

**Comment by sonakravtsenko3** [comment]
Thank you!

Sure, this is something on our mind, too. We can not give a time estimate, but I think this will be done.

**Reply by github-nicolas-crovatti** [reply]
Thanks for your reply ! 

If I were to engage in this task, could you offer some guidance to start on the right foot ?

**Reply by sonakravtsenko3** [reply]
Currently API is being worked on by @graemeniedermayer here: https://github.com/thygate/stable-diffusion-webui-depthmap-script/pull/309, the general structure of the API is still rapidly changing. The file that is responsible for the API is `depthmap_api.py`. It would be a good idea to move it to src folder and make it not depend on gradio.

**Reply by github-nicolas-crovatti** [reply]
Thanks for the hints !

**Comment by graemeniedermayer** [comment]
> It would be a good idea to move it to src folder and make it not depend on gradio.

I think I'll move most of the code to an api folder. I do think the extension api is going to need gradio so everything is coming from the same auto1111 port, but the standalone api could have the gradio component removed. 

gradio does offer a really convenient --share option. So I don't even know if it would be in our best interest. It's basically a temporary SSL proxy for free which is a pretty nice feature. Setting up proxies is kind of a headache

Also if anyone does have any api requests or ideas let me know.

**Reply by AIjie-sen** [reply]
I want to use depth map through the API, can you explain how to use it?

**Reply by sonakravtsenko3** [reply]
@AIjie-sen https://github.com/thygate/stable-diffusion-webui-depthmap-script/issues/313
